### ISSUE TITLE
 [YSQL] signal 11, Segmentation fault - yb::Status::file_name_duplicated

### ISSUE THREAD

**Issue by kmuthukk** [YugaByte/yugabyte-db#1441]
Version: 1.2.9 (Linux)

Test Case:

https://gist.github.com/kmuthukk/310a2199c8d0f8d2647510a1b47442f6

Seeing a new error (perhaps recent regression) when running this test in 1.2.9 release:

`psycopg2.InterfaceError: cursor already closed`

due to a seg fault in the backend:

```
$ gdb /home/centos/yugabyte-1.2.9.0/postgres/bin/postgres core.2767
...
(gdb) where
#0  yb::Status::file_name_duplicated (this=this@entry=0x7f62b4a1a5d0) at ../../src/yb/util/status.cc:239
#1  0x00007f62cdb563e2 in yb::Status::CloneAndAppend (this=this@entry=0x7f62b4a1a5d0, msg=...) at ../../src/yb/util/status.cc:227
#2  0x00007f62cd7723f1 in yb::pggate::PgSession::CombineErrorsToStatus (this=<optimized out>, errors=..., status=...) at ../../src/yb/yql/pggate/pg_session.cc:518
#3  0x00007f62cd772a19 in operator() (status=..., __closure=0x1d6be40) at ../../src/yb/yql/pggate/pg_session.cc:476
#4  boost::detail::function::void_function_obj_invoker1<yb::pggate::PgSession::PgFlushAsync(yb::StatusFunctor)::<lambda(const yb::Status&)>, void, const yb::Status&>::invoke(boost::detail::function::function_buffer &, const yb::Status &) (function_obj_ptr=..., a0=...)
    at /n/jenkins/thirdparty/yugabyte-thirdparty-2019-04-29T16_41_21/thirdparty/installed/uninstrumented/include/boost/function/function_template.hpp:158
#5  0x00007f62c8d2f5b7 in operator() (a0=..., this=<optimized out>)
    at /n/jenkins/thirdparty/yugabyte-thirdparty-2019-04-29T16_41_21/thirdparty/installed/uninstrumented/include/boost/function/function_template.hpp:764
#6  operator() (__closure=<optimized out>) at ../../src/yb/client/batcher.cc:226
#7  std::_Function_handler<void(), yb::client::internal::Batcher::RunCallback(const yb::Status&)::<lambda()> >::_M_invoke(const std::_Any_data &) (__functor=...)
    at /home/centos/yugabyte-1.2.9.0/linuxbrew-xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx/Cellar/gcc/5.5.0_4/include/c++/5.5.0/functional:1871
#8  0x00007f62c8d2d053 in operator() (this=0x1d55058)
    at /home/centos/yugabyte-1.2.9.0/linuxbrew-xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx/Cellar/gcc/5.5.0_4/include/c++/5.5.0/functional:2267
#9  Run (this=0x1d55050) at ../../src/yb/util/threadpool.h:479
#10 yb::client::internal::Batcher::RunCallback (this=this@entry=0x1c64f20, status=...) at ../../src/yb/client/batcher.cc:228
#11 0x00007f62c8d2ea0e in yb::client::internal::Batcher::CheckForFinishedFlush (this=0x1c64f20) at ../../src/yb/client/batcher.cc:221
#12 0x00007f62c8d23db9 in yb::client::internal::AsyncRpc::Finished (this=0x1ee3190, status=...) at ../../src/yb/client/async_rpc.cc:164
#13 0x00007f62c43ae1da in operator() (this=<optimized out>)
    at /home/centos/yugabyte-1.2.9.0/linuxbrew-xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx/Cellar/gcc/5.5.0_4/include/c++/5.5.0/functional:2267
#14 yb::rpc::OutboundCall::InvokeCallbackSync (this=0x1ea3b90) at ../../src/yb/rpc/outbound_call.cc:356
#15 0x00007f62c43f8a74 in yb::rpc::(anonymous namespace)::Worker::Execute (this=<optimized out>) at ../../src/yb/rpc/thread_pool.cc:100
#16 0x00007f62cdb66af9 in operator() (this=0x1c66808)
    at /home/centos/yugabyte-1.2.9.0/linuxbrew-xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx/Cellar/gcc/5.5.0_4/include/c++/5.5.0/functional:2267
#17 yb::Thread::SuperviseThread (arg=<optimized out>) at ../../src/yb/util/thread.cc:669
#18 0x00007f62cd301694 in start_thread (arg=0x7f62b4a1b700) at pthread_create.c:333
#19 0x00007f62cca3e41d in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:109
```

**Comment by kmuthukk** [comment]
There is an open issue with this same test already reported in:

    https://github.com/YugaByte/yugabyte-db/issues/1276

but this particular seg fault is a newer one.

**Comment by rajukumaryb** [comment]
Fixed by @hectorgcr  as part of #1002
